File: packages/cli/src/exporter.js
import fs from 'fs';
import path from 'path';
import ignore from 'ignore';
import { loadData } from './core.js';

// Regex an toàn (dùng negative lookbehind (?<!:) để không xóa nhầm http://)
const REGEX_C_COMMENT = /(?<!:)\/\/.*|\/\*[\s\S]*?\*\//g;
const REGEX_HASH_COMMENT = /^\s*#.*$/gm;
const REGEX_HTML_COMMENT = /<!--[\s\S]*?-->/g;
const REGEX_LOGS = /^\s*(console\.(log|warn|error|info|debug)|print|println!)\s*\(.*?\);?\s*$/gm;

export const exportGroup = (projectPath, groupId, outPath, options) => {
    const data = loadData(projectPath);
    const group = data.groups.find(g => g.id === groupId);
    if (!group) throw new Error("Group không tồn tại");

    const ig = ignore();
    const gitignorePath = path.join(projectPath, '.gitignore');
    if (fs.existsSync(gitignorePath)) ig.add(fs.readFileSync(gitignorePath, 'utf8'));

    let finalContent = `CONTEXT TỪ GROUP: ${group.name}\n`;
    finalContent += `================================================\n\n`;

    const getFiles = (dir) => {
        let results = [];
        if (!fs.existsSync(dir)) return results;
        if (fs.statSync(dir).isFile()) return [dir];
        fs.readdirSync(dir).forEach(file => {
            if (file === '.git' || file === 'node_modules') return;
            const full = path.join(dir, file);
            const rel = path.relative(projectPath, full);
            if (!ig.ignores(rel)) {
                if (fs.statSync(full).isDirectory()) results = results.concat(getFiles(full));
                else results.push(full);
            }
        });
        return results;
    };

    const allFiles = new Set();
    group.paths.forEach(p => getFiles(path.join(projectPath, p)).forEach(f => allFiles.add(f)));

    for (const file of allFiles) {
        try {
            let content = fs.readFileSync(file, 'utf-8');
            if (content.indexOf('\0') !== -1) continue; // Skip binary

            const ext = path.extname(file).toLowerCase();
            const relPath = path.relative(projectPath, file);

            if (options.noLogs) content = content.replace(REGEX_LOGS, '');
            if (options.noComments) {
                if (['.js', '.ts', '.jsx', '.tsx', '.c', '.cpp', '.java', '.cs', '.go', '.rs'].includes(ext)) {
                    content = content.replace(REGEX_C_COMMENT, '');
                } else if (['.py', '.sh', '.yaml', '.yml', '.rb'].includes(ext)) {
                    content = content.replace(REGEX_HASH_COMMENT, '');
                } else if (['.html', '.xml', '.vue'].includes(ext)) {
                    content = content.replace(REGEX_HTML_COMMENT, '');
                }
                // Xóa dòng trống thừa
                content = content.split('\n').filter(line => line.trim().length > 0).join('\n');
            }

            finalContent += `--- FILE: ${relPath} ---\n`;
            if (options.lineNumbers) {
                const lines = content.split('\n');
                finalContent += lines.map((l, i) => `${String(i + 1).padStart(4, ' ')} | ${l}`).join('\n') + '\n';
            } else {
                finalContent += content + '\n';
            }
            finalContent += '\n';

        } catch (e) { /* skip */ }
    }

    fs.writeFileSync(outPath, finalContent);
};

File: packages/cli/src/exporter.rs
// FILE: packages/cli/src/exporter.rs
use crate::core_logic::load_data;
use std::fs;
use std::path::Path;
use ignore::WalkBuilder;
use regex::Regex;
use lazy_static::lazy_static;

lazy_static! {
    static ref C_COMMENT: Regex = Regex::new(r"(?m)^[ \t]*//.*$|(?s)/\*.*?\*/").unwrap();
    static ref HASH_COMMENT: Regex = Regex::new(r"(?m)^[ \t]*#.*$").unwrap();
    static ref HTML_COMMENT: Regex = Regex::new(r"(?s)<!--.*?-->").unwrap();
    static ref DEBUG_LOG: Regex = Regex::new(r"(?im)^\s*(?:console\.(log|warn|error|info|debug)|print|println!|dbg!)\s*\(.*\);?\s*\r?\n?").unwrap();
}

pub struct ExportSettings {
    pub no_comments: bool,
    pub no_logs: bool,
    pub line_numbers: bool,
}

pub fn export_group(project_path: &str, group_id: &str, output_path: &str, settings: ExportSettings) -> Result<(), String> {
    let data = load_data(project_path);
    let group = data.groups.iter().find(|g| g.id == group_id).ok_or("Group không tồn tại")?;
    
    let root = Path::new(project_path);
    let mut final_content = format!("CONTEXT TỪ GROUP: {} (ID: {})\n", group.name, group.id);
    final_content.push_str("================================================\n\n");

    for path_str in &group.paths {
        let target_path = root.join(path_str);
        if !target_path.exists() { continue; }

        for result in WalkBuilder::new(&target_path).build().filter_map(Result::ok) {
            if result.metadata().map(|m| m.is_file()).unwrap_or(false) {
                if let Ok(mut content) = fs::read_to_string(result.path()) {
                    let rel_path = result.path().strip_prefix(root).unwrap().to_string_lossy();
                    
                    if settings.no_logs {
                        content = DEBUG_LOG.replace_all(&content, "").to_string();
                    }
                    if settings.no_comments {
                        let ext = result.path().extension().and_then(|s| s.to_str()).unwrap_or("");
                        content = match ext {
                            "js" | "ts" | "jsx" | "tsx" | "rs" | "go" | "c" | "cpp" | "java" | "cs" => C_COMMENT.replace_all(&content, "").to_string(),
                            "py" | "sh" | "yaml" | "yml" | "rb" => HASH_COMMENT.replace_all(&content, "").to_string(),
                            "html" | "xml" | "md" => HTML_COMMENT.replace_all(&content, "").to_string(),
                            _ => content
                        };
                        // Xóa các dòng trống thừa do xóa comment
                        content = content.lines().filter(|l| !l.trim().is_empty()).collect::<Vec<_>>().join("\n");
                    }

                    final_content.push_str(&format!("--- FILE: {} ---\n", rel_path));
                    if settings.line_numbers {
                        for (i, line) in content.lines().enumerate() {
                            final_content.push_str(&format!("{:>4} | {}\n", i + 1, line));
                        }
                    } else {
                        final_content.push_str(&content);
                        final_content.push('\n');
                    }
                    final_content.push_str("\n\n");
                }
            }
        }
    }

    fs::write(output_path, final_content).map_err(|e| e.to_string())?;
    Ok(())
}
